ACViewer-SDK/Assets/SkinsModule/SkinController.cs

```diff
@@ -0,0 +1,304 @@
+        // Default adress: https://developer.augmented.city/media/resources/ACV/props_by_gps.json
+        yield return sw.SendWebRequest();
+        if (sw.isNetworkError || sw.isHttpError)
+        {
+            Debug.Log(sw.error);
+        }
+        else
+        {
+            getJsonParams(sw.downloadHandler.text);
+            Debug.Log("GetUIParams Json = " + sw.downloadHandler.text);
+        }
+    }
+
+    public void ReloadScene()
+    {
+        SceneManager.LoadScene(0);
+    }
+
+    void UIParamsParcer(string jsonS)
+    {
+        var jsonParse = JSON.Parse(jsonS);
+        string pl, sk, ni, na, la, lo, ra;
+        int objectsAmount = -1;
+        List<PlaceInfo> placeInfoList = new List<PlaceInfo>();
+        do
+        {
+            objectsAmount++;
+            pl = jsonParse[objectsAmount]["place"];
+            sk = jsonParse[objectsAmount]["skin"];
+            ni = jsonParse[objectsAmount]["no_instructions"];
+            na = jsonParse[objectsAmount]["no_start-AR"];
+            la = jsonParse[objectsAmount]["zone"]["latitude"];
+            lo = jsonParse[objectsAmount]["zone"]["longitude"];
+            ra = jsonParse[objectsAmount]["zone"]["radius_km"];
+            if (pl != null)
+            {
+                PlaceInfo newPlace = new PlaceInfo(pl, sk, ni, na, la, lo, ra);
+                placeInfoList.Add(newPlace);
+            }
+            Debug.Log("UIParamsParcer node [" + objectsAmount + "]  - " + pl);
+        } while (pl != null);
+        Debug.Log("placeInfoList.Count  [ ==" + placeInfoList.Count);
+        CheckLocation(placeInfoList.ToArray());
+    }
+
+    public void CheckLocation(PlaceInfo[] places)
+    {
+        PlaceInfo rightPlace = null;
+        PlaceInfo allPlaces = null;
+        System.Threading.Thread.CurrentThread.CurrentCulture = new System.Globalization.CultureInfo("en-US");
+        for (int i = 0; i < places.Length; i++) 
+        {
+            if (places[i].place.ToLower().Contains("all other places"))
+            {
+                allPlaces = places[i];
+            }
+            else
+            {
+                float lati = 0;
+                float.TryParse(places[i].zone.latitude, out lati);
+                float longi = 0;
+                float.TryParse(places[i].zone.longitude, out longi);
+                float radiusKm = 0;
+                float.TryParse(places[i].zone.radiusKm, out radiusKm);
+                Debug.Log("Place "+places[i].place +", point # = " + i +", geo: la = " + lati + ", lo = " + longi + ", radKm = " + radiusKm);
+                if (DistanceTo(lat, lon, lati, longi) < radiusKm)
+                {
+                    rightPlace = places[i];
+                    Debug.Log("Place location = " + rightPlace.place + ", point # = " + i);
+                }
+            }
+        }
+
+        if (rightPlace != null) 
+        {
+            SetPlace(rightPlace);
+        }
+        else
```

Comment:
Bad condition...:
NotInPlace = false;  -> move to the state of the class!
if (right) { setPlace(right); } else if (alloth) { setPlace(alloth); } else if (places.Length >= 1) { NotInPlace=true; }
....
if (NotInPlace) { Panel.active.inPlace(1) }